app/src/main/java/org/mozilla/focus/FocusApplication.kt

```diff
@@ -39,11 +45,31 @@ class FocusApplication : LocaleAwareApplication() {
 
         SearchEngineManager.getInstance().init(this)
         TelemetryWrapper.init(this)
+        initGlean()
         DataUploadPreference.init(this.applicationContext)
 
         registerActivityLifecycleCallbacks(VisibilityLifeCycleCallback(this))
     }
 
+    private fun initGlean() {
+        DataUploadUpdateEvent.dataUploadLiveData.observeForever { uploadEnabled ->
+            Glean.setUploadEnabled(uploadEnabled)
+        }
+
+        val telemetryKey = applicationContext.getString(PREF_KEY_TELEMETRY)
+        val initialUploadEnabled =
+            PreferenceManager.getDefaultSharedPreferences(this).getBoolean(telemetryKey, false)
```

Comment:
Apologies if I'm missing something basic here, but how is this value set in SP?  Also, is there any reason we can't use `initialUploadEnabled` as the key instead of the telemetry code?